maven-plugin-api/src/main/java/org/apache/maven/plugin/descriptor/PluginDescriptor.java

```diff
@@ -49,6 +50,8 @@ public class PluginDescriptor extends ComponentSetDescriptor implements Cloneabl
 
     private static final String LIFECYCLE_DESCRIPTOR = "META-INF/maven/lifecycle.xml";
 
+    private static final Pattern PATTERN_FILTER_1 = Pattern.compile("(-?maven-?)|(-?plugin-?)");
```

Comment:
Why so complex? `-?(maven|plugin)-?`, no?

Reply:
You are right, my target was to migrate in Patterns firstly

Reply:
This I understood as well, but reduction is always good.

Reply:
"-?(mave)|(plugi)n-?" ))

Reply:
Technically, yes, but reads very very badly